Bump angular to v14

oliverguenther opened the pull request:
Bumps Angular to v14

- [x] Fixes strict mode import errors on legacy javascript
- [x] Replace or fix incompatible modules that got vendored
- [x] Remove delayed bootstrap which is no longer necessary

oliverguenther committed:
Remove delayed bootstrap

oliverguenther committed:
fix reference error with newer TS compile

oliverguenther committed:
Bump angular to v14

oliverguenther committed:
Replace qrcode with package

oliverguenther committed:
Fix language in non-headless option

oliverguenther committed:
Fix JS errors on reporting page

oliverguenther committed:
Work around fullcalendar css import

https://github.com/fullcalendar/fullcalendar-angular/issues/403

oliverguenther committed:
Fix enjoyhint references to window

oliverguenther committed:
Fix type errors after rebase

oliverguenther committed:
Fix backup component being disabled by verification

oliverguenther committed:
Fix member form

oliverguenther committed:
Fix trailing slash redirect, this breaks onboarding tour by not taking
over the query param

oliverguenther committed:
Fix script errors and make spec wait better

oliverguenther committed:
Fix waitForElement if element is already present

oliverguenther committed:
Fix missing const in project script

b12f commented on frontend/src/app/features/user-preferences/notifications-settings/page/notifications-settings-page.component.ts:
```diff
@@ -201,24 +201,24 @@ export class NotificationsSettingsPageComponent extends UntilDestroyedMixin impl
+            workPackageCommented: new UntypedFormControl(setting.workPackageCommented),
+            startDate: new UntypedFormControl(setting.startDate),
+            dueDate: new UntypedFormControl(setting.dueDate),
+            overdue: new UntypedFormControl(setting.overdue),
```
I might be misunderstanding this, but don't we know the exact types of the form groups & controls here?

oliverguenther replied:
That's just the angular migration running and transforming it automatically. I haven't looked into detail on how we can type it.

b12f reviewed (approved):
LGTM, thanks for doing this

oliverguenther committed:
Redirect with query params in overview module

oliverguenther committed:
Rewrite cost spec for better waiting

oliverguenther committed:
Improve board sorting waiting

oliverguenther committed:
Restore current bootstrap, it breaks some services

some controllers access dom nodes on their constructor, which breaks with

the replaced method

oliverguenther committed:
Write gon also on create response
